app/cdap/components/ConnectionList/index.tsx

```diff
@@ -199,32 +199,40 @@ export default function ConnectionList() {
     );
   };
 
+  const setSnackbarState = (value: boolean) => {
+    setSnackbar({
+      open: false,
+    });
+  };
+
   let headerContent;
 
   return (
     <Box data-testid="data-sets-parent" className={classes.connectionsListContainer}>
       <SubHeader />
       <SelectDatasetWrapper>
-        {dataForTabs.map((each, index) => {
-          const connectionIdRequired = each.data.filter((el) => el.connectionId);
-          if (connectionIdRequired.length) {
-            connectionId = connectionIdRequired[0].connectionId;
+        {dataForTabs?.map((eachTabItem, tabIndex) => {
+          const requiredConnectionId = eachTabItem?.data?.filter(
+            (eachDataItem) => eachDataItem.connectionId
+          );
+          if (requiredConnectionId.length) {
+            connectionId = requiredConnectionId[0].connectionId;
           }
-          if (index === 0) {
+          if (tabIndex === 0) {
             headerContent = headerForLevelZero();
           } else {
             headerContent =
-              refs.current[index]?.offsetWidth < refs.current[index]?.scrollWidth ? (
-                <CustomTooltip title={dataForTabs[index - 1].selectedTab} arrow>
+              refs.current[tabIndex]?.offsetWidth < refs.current[tabIndex]?.scrollWidth ? (
```

Comment:
Please don't have big blocks of code in ternary operators

Reply:
This connection list code refactoring will be covered as part of the connection list feature PR
currently not updating here, but will be taken for connection list feature PR